linera-ethereum/src/test_utils/mod.rs

```diff
@@ -30,12 +43,115 @@ sol!(
     "./contracts/EventNumerics.json"
 );
 
+#[allow(clippy::type_complexity)]
+#[derive(Clone)]
+pub struct EthereumClient {
+    pub provider: FillProvider<
```

Comment:
nit: Maybe extract to `pub type EthereumClientProvider = ` and reuse `EthereumClientProvider` here?

Reply:
The problem is that this prevents implementing a new operator. Because we would be implementing function for a function defined out of crate.